### ISSUE TITLE
NRE in `ClickHouseConnection.Dispose` after 2.2.10 update

### ISSUE THREAD

**Issue by MaceWindu** [Octonica/ClickHouseClient#70]
Note that crash happens in `Finalizer` thread and crash application currently

BTW, it would be nice to publish `snuget` symbols package

![image](https://user-images.githubusercontent.com/1099238/210140185-002a89d3-0f4d-42d3-8f04-88f80e464d3b.png)

**Comment by MaceWindu** [comment]
Looks like double disposal to me. Currently Dispose in `ClickHouseConnection` doesn't call `SupporessFinalize`

**Comment by MaceWindu** [comment]
What is strange - I don't see any changes that could cause it. Probably issue existed before and just very rare

**Comment by victor-sushko** [comment]
> Looks like double disposal to me. Currently Dispose in `ClickHouseConnection` doesn't call `SupporessFinalize`

`GC.SuppressFinalize(this)` is called in `Dispose` from the base class `System.ComponentModel.Component`. Probably, the connection was not disposed properly.

However, `_connectionState` should never be `null`, even when the connection object was disposed. And especially when the connection was not disposed.

Could you please provide more steps for reproducing this error?

**Comment by MaceWindu** [comment]
Will check a bit later

**Comment by MaceWindu** [comment]
Actually that's regression in last version. When incorrect connection string passed to connection constructor, connection initialization doesn't complete properly:

```cs
using Octonica.ClickHouseClient;

internal class Program
{
    private static void Main()
    {
        Test();

        // trigger finalizers
        while (true)
        {
            GC.Collect();
        }
    }

    private static void Test()
    {
        try
        {
            // dispose never called due to .ctor exception
            using var _ = new ClickHouseConnection("BAD");
        }
        catch
        {
        }
    }
}
```

**Comment by SergeyMirvoda** [comment]
@MaceWindu wow, this was unexpected)
thank you!
@victor-sushko can we fix this ASAP?

**Comment by victor-sushko** [comment]
This bug was fixed in [v2.1.11](https://www.nuget.org/packages/Octonica.ClickHouseClient/2.2.11).
